test/lib/commands/init.js

```diff
@@ -540,3 +569,33 @@ t.test('npm init workspces with root', async t => {
   t.equal(pkg.license, 'ISC')
   t.matchSnapshot(npm._mockOutputs, 'does not print helper info')
 })
+
+t.test('no args, missing package', async t => {
+  t.teardown(() => {
+    npm._mockOutputs.length = 0
+  })
+
+  const Init = t.mock('../../../lib/commands/init.js', {
+    ...mocks,
+    'proc-log': {
+      ...mocks['proc-log'],
+      warn: (msg) => {
+        t.equal(msg, 'Missing package.json. Try with `--include-workspace-root`.')
+      },
+    },
+  })
+  const init = new Init(npm)
+  // init-package-json prints directly to console.log
+  // this avoids poluting test output with those logs
+  console.log = noop
+
+  npm.localPrefix = t.testdir({})
+
+  await t.rejects(
+    init.execWorkspaces([], ['a']),
+    { code: 'ENOENT' },
+    'should exit with missing package.json file error'
+  )
+
+  t.matchSnapshot(npm._mockOutputs, 'should print warn info')
```

Comment:
this snapshot is empty because of the mock above that asserts on the log.warn call. it doesn't end up in the mock outputs because of that change so we should either remove this snapshot or make the mock correctly append to the mock outputs